drfilters/filter.go

```diff
@@ -0,0 +1,45 @@
+package drfilters
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	None         = 0
+	Mode         = 1
+	StdDeviation = 2
+)
+
+type (
+	tallyProp struct {
```

Comment:
It's better to have a struct for each filter type instead of doing optional. And have a prestep where you decode the rlp and switch case on the mode to get the correct struct

Reply:
The approach you mentioned is difficult to implement. It's not possible to know the filtering mode, without decoding to the struct first.

Reply:
In rust we decode by hand, couldn't you get the first index of RLP and then decode into a struct?
The thing is RLP is indexed based and if we have different properties in the future then we have to zero out optional properties on a protocol level which seems weird to do just for Go, while other languages handle this fine.

For example in rust:
```
fn extract_bytes(tokens: &[Token], index: usize, name: &str) -> Result<Vec<u8>, RelayerError> {
    extract_token(tokens, index, name, |t| match t {
        Token::FixedBytes(bytes) => Some(bytes),
        other => other.into_bytes(),
    })
}
```

Reply:
here: [c015652](https://github.com/sedaprotocol/seda-chain/pull/280/commits/c015652b646e34e940d39e677ac65988057a6a61)